VoodooPS2Trackpad/VoodooPS2Elan.h

```diff
@@ -0,0 +1,365 @@
+/*
+ * Elan PS2 touchpad integration
+ *
+ * Mostly contains code ported from Linux
+ * https://github.com/torvalds/linux/blob/master/drivers/input/mouse/elantech.c
+ *
+ * Created by Bartosz Korczyński (@bandysc), Hiep Bao Le (@hieplpvip)
+ * Special thanks to Kishor Prins (@kprinssu), EMlyDinEsHMG and whole VoodooInput team
+ */
+
+#ifndef _APPLEPS2ELAN_H
+#define _APPLEPS2ELAN_H
+
+#include "../VoodooPS2Controller/ApplePS2MouseDevice.h"
+#include "LegacyIOHIPointing.h"
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winconsistent-missing-override"
+#include <IOKit/IOCommandGate.h>
+#include <IOKit/acpi/IOACPIPlatformDevice.h>
+#pragma clang diagnostic pop
+
+#include "VoodooInputMultitouch/VoodooInputEvent.h"
+
+struct virtual_finger_state {
+    TouchCoordinates prev;
+    TouchCoordinates now;
+    uint8_t pressure;
+    uint8_t width;
+    bool touch;
+    bool button;
+    MT2FingerType fingerType;
+};
+
+typedef enum {
+    FORCE_TOUCH_DISABLED = 0,
+    FORCE_TOUCH_BUTTON = 1,
+    FORCE_TOUCH_THRESHOLD = 2,
+    FORCE_TOUCH_VALUE = 3,
+    FORCE_TOUCH_CUSTOM = 4
+} ForceTouchMode;
```

Comment:
This should be moved to some common header

Reply:
Should we do it in this pull request or just merge and do it later?

Reply:
I would say could do here.